spec/support/html_validator.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Use Nokogiri to validate HTML
+RSpec::Matchers.define :be_valid_html do
+  match do |actual_html|
+    Nokogiri::HTML5(actual_html).errors.empty?
```

Comment:
Have you verified that this works as intended? At least according to https://nokogiri.org/rdoc/Nokogiri/HTML5.html "by default, no parse errors are reported".

Also, are you sure that this catches the sort of HTML5 errors that the W3C validator does? If this doesn't do much more than catch XML errors, not sure it is worth the overhead.

Reply:
@justinlittman Good catch. I removed the `max_errors` param in a refactoring without knowing what it did. 😅 It's back now.

This would catch any errors encountered when using this parser: https://github.com/sparklemotion/nokogiri/tree/main/gumbo-parser/src

Not sure how that stacks up to the W3C validator. There's not a lot of code here we'd need to throw out, at least, if we come up a superior way to automate validation of app HTML.